mypy/typeanal.py

```diff
@@ -232,7 +232,9 @@ def visit_unbound_type(self, t: UnboundType) -> Type:
                     # context. This is slightly problematic as it allows using the type 'Any'
                     # as a base class -- however, this will fail soon at runtime so the problem
                     # is pretty minor.
-                    return AnyType()
+                    any_type = AnyType()
+                    any_type.is_from_silent_import = sym.node.type.is_from_silent_import
```

Comment:
This suggests `is_from_silent_import` to me that should be an optional argument to `AnyType`'s constructor.